### PR TITLE
Copy propagation tweaking

### PR DESCRIPTION
**Pull request by SingleAccretion**
Some small CQ improvements in preparation for a zero-diff TP-oriented refactoring.

Please refer to the individual commit messages for details.

[Diffs](https://dev.azure.com/dnceng/public/_build/results?buildId=1599841&view=ms.vss-build-web.run-extensions-tab).

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib

**Review by TIHan — APPROVED** [review]
LGTM

File: src/coreclr/jit/compiler.h [hunk 1]
```diff
@@ -10858,6 +10858,8 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 public:
     struct ShadowParamVarInfo
     {
+        static const unsigned NO_SHADOW_COPY = UINT_MAX;
```
Comment (AndyAyersMS) [inline src/coreclr/jit/compiler.h:10861]: Nit: why not just use `BAD_VAR_NUM`?

Comment (SingleAccretion) [reply src/coreclr/jit/compiler.h]: I suppose no reason. Deleted it.

**Review by AndyAyersMS — APPROVED** [review]
Thanks again for your ongoing contributions.

**Pull request merged**